framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -203,6 +212,29 @@ MeshGenerator::declareMeshProperty(const std::string & data_name, const T & init
   return data;
 }
 
+template <typename T>
+T &
+MeshGenerator::declareMeshProperty(const std::string & mg_name,
+                                   const std::string & data_name,
+                                   const T & init_value)
+{
+  std::string full_name =
+      std::string(MeshMetaDataInterface::SYSTEM) + "/" + mg_name + "/" + data_name;
+
+  // Here we will create the RestartableData even though we may not use this instance.
+  // If it's already in use, the App will return a reference to the existing instance and we'll
+  // return that one instead. We might refactor this to have the app create the RestartableData
+  // at a later date.
+  auto data_ptr = std::make_unique<RestartableData<T>>(full_name, nullptr);
+  auto & restartable_data_ref = static_cast<RestartableData<T> &>(_app.registerRestartableData(
+      full_name, std::move(data_ptr), 0, true, MooseApp::MESH_META_DATA));
+
+  T & data = restartable_data_ref.set();
+  data = init_value;
+
```

Comment:
also isnt there concerns if the mesh generator has already be ran?

Reply:
For some context, this function basically allows you to overwrite the value of `data_name` from another mesh generator. I wasn't sure which checks I should add to allow such behavior but this function would basically be used to update the values of parameters of mesh generators that have already been called from an external mesh generator, since to my knowledge this functionality is not available in MOOSE MG's and we need a way to keep track of block id's that have already been assigned at each stage of PinMeshGenerator, AssemblyMeshGenerator, and CoreMeshGenerator. This block ID mapping is being stored by an intermediate MG ReactorMeshParams, and each of PMG, AMG, and CMG will query ReactorMeshParams for the block ID mapping, then update this block ID mapping after their respective constructors and `generate` functions have been called.

Reply:
This is now removed